src/frontend/components/bigHorizontalVideoCard/bigHorizontalVideoCard.js

```diff
@@ -0,0 +1,148 @@
+import { useNavigate } from "react-router-dom";
+import { useState } from 'react';
+import "./bigHorizontalVideoCard.css";
+import { addToLikeService , removeFromLikeService, addToWatch, removeFromWatch} from "../../services/"
+import { useLikedVideos , useAuth, useWatchLaterVideos} from "../../contexts"
+
+const BigHorizontalCard =({video})=>{
+
+    const navigate= useNavigate();
+    const { stateAuth:{token, isLoggedIn}}= useAuth();
+    const { likedVideos, setLikedVideos} = useLikedVideos();
+    const { watchLaterVideos,setWatchLaterVideos }= useWatchLaterVideos();
+
+     const [showModal, setShowModal] = useState(false);
+
+    const toggleModal=()=>{
+        setShowModal((showModal)=>!showModal)
+    }
+
+    const isInWatchLater = watchLaterVideos.find(watchLaterVideo=>watchLaterVideo._id===video._id ? true : false)
+    const isInLikedVideos = likedVideos.find(likedVideo=>likedVideo._id===video._id ? true : false)
+
+    
+    const addToWatchLaterVideos = async()=>{
+        if(isLoggedIn){
+            try{
+                const response = await addToWatch(video,token)
+                console.log(response)
+                if(response.status===201){
+                    setWatchLaterVideos(response.data.watchlater)
+                }
+        
+            }catch(error){
+                console.log(error);
+            }
+
+        }else{
+            navigate("/login")
+        }
+           
+    }
+
+    const addToLikedVideos = async()=>{
+
+        try{
+            const response = await addToLikeService(video, token)
+            if(response.status===201){
+                setLikedVideos(response.data.likes)
+            }
+        }
+        catch(error){
+            console.error(error)
```

Comment:
try using some toasts library . this shows foresight for interviewer. 
if backed breaks, frontend isnt blank , it shows error thats good UX